core/lib/src/server.rs

```diff
@@ -94,14 +99,20 @@ async fn hyper_service_fn(
             }
         };
 
-        // Retrieve the data from the hyper body.
-        let mut data = Data::from(h_body);
+        let span = info_span!("request", "{}", req);
+        async {
+            // Retrieve the data from the hyper body.
+            let mut data = Data::from(h_body);
 
-        // Dispatch the request to get a response, then write that response out.
-        let token = rocket.preprocess_request(&mut req, &mut data).await;
-        let r = rocket.dispatch(token, &mut req, data).await;
-        rocket.send_response(r, tx).await;
-    });
+            // Dispatch the request to get a response, then write that response out.
+            let token = rocket.preprocess_request(&mut req, &mut data).await;
+            let r = rocket.dispatch(token, &mut req, data).await;
+            rocket.send_response(r, tx).await;
+        }
+            .instrument(span)
+            .await
+
+    }.in_current_span());
```

Comment:
Do we need both of these?

Reply:
The `instrument` call is what set ups the `Request:` line in the log (and indents everything under it).

The `in_current_span` is there for the same reason as the other call to it.